src/network/mod.rs

```diff
@@ -0,0 +1,154 @@
+mod smol_executor;
+
+use std::net::TcpListener;
+
+use crate::structs::{OutputValues, Clients, Arm};
+
+use self::smol_executor::*;
+use async_compat::*;
+use futures_util::sink::SinkExt;
+use smol::{prelude::*, Async};
+
+use async_tungstenite::{
+    tungstenite::{handshake::derive_accept_key, protocol::Role, Message}, 
+    WebSocketStream
+};
+
+use eyre::{Result, Error};
+use hyper::{
+    Server, service::{make_service_fn, service_fn}, 
+    Request, Body, Response, StatusCode, 
+    header::{
+        HeaderValue, 
+        CONNECTION, UPGRADE, SEC_WEBSOCKET_ACCEPT, SEC_WEBSOCKET_KEY}
+};
+
+
+pub async fn handle_clients(values: Arm<OutputValues>, clients: Clients) {
+    let _span = tracy_client::span!("handle clients");
+    let mut clients = clients.lock().unwrap();
```

Comment:
To keep the `MutexGuard`'s scope as small as possible, acquiring it should happen after the `serde_json` business.